crates/mpc-attestation/src/attestation.rs

```diff
@@ -0,0 +1,207 @@
+                    allowed_mpc_docker_image_hashes,
+                    allowed_launcher_docker_compose_hashes,
+                    timestamp_seconds,
+                );
+            }
+        };
+
+        // Embedded JSON assets
+        const TCB_INFO_STRING_PROD: &str = include_str!("../assets/tcb_info.json");
+        // TODO Security #1433 - remove dev measurements from production builds after testing is complete.
+        const TCB_INFO_STRING_DEV: &str = include_str!("../assets/tcb_info_dev.json");
+
+        let expected_measurements_list = ExpectedMeasurements::from_embedded_tcb_info(&[
+            TCB_INFO_STRING_PROD,
+            TCB_INFO_STRING_DEV,
+        ])
+        .map_err(VerificationError::EmbeddedMeasurementsParsing)?;
+
+        attestation.verify(
+            expected_report_data,
+            timestamp_seconds,
+            &expected_measurements_list,
+        )
+    }
+
+    /// Verifies MPC node image hash is in allowed list.
+    fn verify_mpc_hash(
+        &self,
+        tcb_info: &TcbInfo,
+        allowed_hashes: &[MpcDockerImageHash],
+    ) -> Result<(), VerificationError> {
+        let event = tcb_info.get_single_event(MPC_IMAGE_HASH_EVENT)?;
+
+        allowed_hashes
+            .iter()
+            .any(|hash| hash.as_hex() == *event.event_payload)
+            .or_err(|| {
+                VerificationError::Custom(format!(
+                    "MPC image hash {} is not in the allowed hashes list",
+                    event.event_payload.clone()
+                ))
+            })
+    }
+
+    fn verify_launcher_compose_hash(
+        &self,
+        tcb_info: &TcbInfo,
+        allowed_hashes: &[LauncherDockerComposeHash],
+    ) -> Result<(), VerificationError> {
+        let app_compose: AppCompose = serde_json::from_str(&tcb_info.app_compose)
+            .map_err(|e| VerificationError::AppComposeParsing(e.to_string()))?;
+
+        let launcher_bytes: [u8; 32] =
+            Sha256::digest(app_compose.docker_compose_file.as_bytes()).into();
+
+        allowed_hashes
+            .iter()
+            .any(|hash| hash.deref() == &launcher_bytes)
+            .or_err(|| {
+                VerificationError::Custom(format!(
+                    "launcher compose hash {} is not in the allowed hashes list",
+                    hex::encode(launcher_bytes.as_ref(),)
+                ))
+            })
+    }
+}
+
+#[derive(Debug, Default, Clone, Serialize, Deserialize, BorshDeserialize, BorshSerialize)]
+pub enum MockAttestation {
+    #[default]
+    /// Always pass validation
+    Valid,
+    /// Always fails validation
+    Invalid,
+    /// Pass validation depending on the set constraints
+    WithConstraints {
+        mpc_docker_image_hash: Option<MpcDockerImageHash>,
+        launcher_docker_compose_hash: Option<LauncherDockerComposeHash>,
+
+        /// Unix time stamp for when this attestation expires.  
+        expiry_time_stamp_seconds: Option<u64>,
+    },
+}
```

Comment:
How will this `MockAttestation` struct play with the `MockAttestation` inside the attestation crate?

https://github.com/near/mpc/blob/2f09f9197b46b96083d63e161444c7f86045114b/crates/attestation/src/attestation.rs#L120-L131

Reply:
atm they dont play. As you noticed the MockAttestation from the attestation crate is not used. Lets continue this in the other comment or in the internal thread